### DISCUSSION TITLE
Really want to learn about the lua type hint system, any recommand resource?

### DISCUSSION THREAD

**Discussion by LintaoAmons** [folke/flash.nvim#107] [Q&A] [answered]
I want write lua plugin and I saw in your plugin there's a lot of type hints.
Not familiar with lua and did some search but didn't found any good resource for beginners~

do you have some recommandation for this?

```lua
---@class Flash.Pattern
---@field pattern string
---@field search string
---@field skip string
---@field trigger string
---@field mode Flash.Pattern.Mode
---@operator call:string Returns the input pattern
local M = {}
M.__index = M
```

```lua
{
  "folke/flash.nvim",
  event = "VeryLazy",
  ---@type Flash.Config
  opts = {},
...
}
```

**Comment by folke** [comment, marked as answer]
You can fine an overview of the lua_ls annotations here https://github.com/LuaLS/lua-language-server/wiki/Annotations

**Reply by LintaoAmons** [reply]
Thx, this doc is fantastic
